src/esrs/state.rs

```diff
@@ -60,6 +62,25 @@ impl<S: Default> AggregateState<S> {
         }
     }
 
+    /// Consumes the aggregate state and generates a new one with the events applied to it,
+    /// as dictaded by the given `apply_events`.
+    pub fn apply_store_events<T, F>(self, store_events: Vec<StoreEvent<T>>, apply_event: F) -> Self
+    where
+        F: Fn(S, T) -> S,
+    {
+        let sequence_number: SequenceNumber = store_events.last().map_or(0, StoreEvent::sequence_number);
+        let inner = store_events
+            .into_iter()
+            .map(|store_event| store_event.payload)
+            .fold(self.inner, apply_event);
+
+        AggregateState {
+            sequence_number,
+            inner,
+            ..self
+        }
+    }
```

Comment:
Moved here from AggregateManager, as this really deals with the internals of AggregateState.
You need to pass the function `(State, Event) -> State` which really is `Aggregate::apply_event`.